test/network/2020-03-01/armnetwork/models.go

```diff
@@ -10148,10 +10148,10 @@ type SessionIDs struct {
 // StringArrayResponse is the response envelope for operations that return a []string type.
 type StringArrayResponse struct {
 	// RawResponse contains the underlying HTTP response.
-	RawResponse *http.Response
+	RawResponse *http.Response `json:"-"`
 
 	// Response for ApplicationGatewayAvailableServerVariables API service call.
-	StringArray *[]string
+	StringArray *[]string `json:"StringArray,omitempty"`
```

Comment:
I believe this should be named `Value` with a tag of the same (that's how it is in track 1).

Reply:
Hmmm let me look at the naming rules/ transform that happens before this. I've seen this sort of naming in several other files.

Reply:
So this is what we do in the transform: 
```
case SchemaType.Array:
      const arraySchema = <ArraySchema>schema;
      const arrayElem = <Schema>arraySchema.elementType;
      return `${recursiveTypeName(arrayElem)}Array`;
```

Reply:
Also we name ByteArrays as `ByteArray`

Reply:
If we want to change this, I would say we should do it in a separate PR since it will impact many generated files

Reply:
For a little more context, in track 1 we referred to the enclosing type (`StringArrayResponse` in this example) as a "synthetic type".  You can find the rules for determining when to create a synthetic type [here](https://github.com/Azure/autorest.go/blob/4842cb44812dfd73786d236d32abc944bf821654/src/Extensions.cs#L380).

Reply:
Ok sounds good, I'll add a similar rule for track2, but I think it would be better to do it separately to not mix it in with these changes.

Reply:
Hmm I think this also gets into a data-plane/management-plane thing.  For data-plane I think the naming as-is makes sense as there's no envelope requirement.  For ARM, naming it `Value` is more inline with their RPC requirements.  At any rate I agree that we can solve that separately.

Reply:
I see, I'll consider different behaviors depending on whether we're generating for ARM or data-plane when I implement it.